blitz_api/tests/tests_view_ResetPassword.py

```diff
@@ -0,0 +1,309 @@
+            "EMAIL_SERVICE": True,
+            "FRONTEND_INTEGRATION": {
+                "FORGOT_PASSWORD_URL": "fake_url",
+            }
+        }
+    )
+    @mock.patch('blitz_api.views.IMailing')
+    def test_create_new_token_without_email_param(self, imailing):
+        """
+        Ensure we can't have a new token to change our password without
+        give our email in param
+        """
+        data = dict()
+
+        response = self.client.post(
+            reverse('reset_password'),
+            data,
+            format='json',
+        )
+
+        # The token has been created
+        tokens = ActionToken.objects.filter(
+            user=self.user,
+            type='password_change',
+        )
+
+        content = {
+            'email': ["This field is required."],
+        }
+        self.assertEqual(json.loads(response.content), content)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        self.assertTrue(len(tokens) == 0)
+
+    @override_settings(
+        LOCAL_SETTINGS={
+            "EMAIL_SERVICE": True,
+            "FRONTEND_INTEGRATION": {
+                "FORGOT_PASSWORD_URL": "fake_url",
+            }
+        }
+    )
+    @mock.patch('blitz_api.views.IMailing')
+    def test_create_new_token_with_an_empty_email_param(self, imailing):
+        """
+        Ensure we can't have a new token to change our password without
+        give our email in param
+        """
+        data = {
+            'email': '',
+        }
+
+        response = self.client.post(
+            reverse('reset_password'),
+            data,
+            format='json',
+        )
+
+        # The token has been created
+        tokens = ActionToken.objects.filter(
+            user=self.user,
+            type='password_change',
+        )
+
+        content = {
+            'email': ["This field may not be blank."],
+        }
+        self.assertEqual(json.loads(response.content), content)
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        self.assertTrue(len(tokens) == 0)
+
+    @override_settings(
+        LOCAL_SETTINGS={
+            "EMAIL_SERVICE": True,
+            "FRONTEND_INTEGRATION": {
+                "FORGOT_PASSWORD_URL": "fake_url",
+            }
+        }
+    )
+    @mock.patch('blitz_api.views.IMailing')
+    def test_create_new_token_with_bad_email(self, imailing):
+        """
+        Ensure we can't have a new token to change our password without
+        a valid email
+        """
+        data = {
+            'email': 'test',
+        }
+
+        response = self.client.post(
+            reverse('reset_password'),
+            data,
+            format='json',
+        )
+
+        # The token has been created
+        tokens = ActionToken.objects.filter(
+            user=self.user,
+            type='password_change',
+        )
+
+        content = {
+            'detail': "No account with this email.",
```

Comment:
May be :
```
{
    'email': "No account with this email."
}
```

Because it's link the the email.
Optionnaly it could be `non_field_error` but it never can be `detail`

Reply:
Agreed, will fix that